.github/workflows/rc.yml

```diff
@@ -281,6 +276,7 @@ jobs:
           set -e
           # make brew Java available to CMake
           export JAVA_HOME=$(brew --prefix openjdk@11)/libexec/openjdk.jdk/Contents/Home
+          export ARROW_BUILD_TESTS=OFF
```

Comment:
I disabled this

(1) we're not running the tests in CI anyways, so we might as well save some time
(2) they fail to build currently due to whatever Boost/Homebrew just did

Reply:
How about disabling tests by default?

```diff
diff --git a/ci/scripts/jni_macos_build.sh b/ci/scripts/jni_macos_build.sh
index 65b255e2..f7543b6f 100755
--- a/ci/scripts/jni_macos_build.sh
+++ b/ci/scripts/jni_macos_build.sh
@@ -61,7 +61,7 @@ github_actions_group_begin "Building Arrow C++ libraries"
 install_dir="${build_dir}/cpp-install"
 : "${ARROW_ACERO:=ON}"
 export ARROW_ACERO
-: "${ARROW_BUILD_TESTS:=ON}"
+: "${ARROW_BUILD_TESTS:=OFF}"
 export ARROW_BUILD_TESTS
 : "${ARROW_DATASET:=ON}"
 export ARROW_DATASET
```